File: Leetcode/706/706.cpp
class MyHashMap {
private:
    static const int MAX_LENGTH = 1000005;
    static const int INF = 0x3f3f3f3f;
    int mArray[MAX_LENGTH];
public:
    /** Initialize your data structure here. */
    MyHashMap() {
        memset(mArray, 0x3f, sizeof mArray);
    }
    
    /** value will always be non-negative. */
    void put(int key, int value) {
        mArray[key] = value;
    }
    
    /** Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key */
    int get(int key) {
        return (mArray[key] == INF) ? -1 : mArray[key];
    }
    
    /** Removes the mapping of the specified value key if this map contains a mapping for the key */
    void remove(int key) {
        mArray[key] = INF;
    }
};

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap* obj = new MyHashMap();
 * obj->put(key,value);
 * int param_2 = obj->get(key);
 * obj->remove(key);
 */

File: Leetcode/706/706.go
const (
	MAX_LENGTH = 1e6 + 5
)

type MyHashMap struct {
	hashmap []int
}

func Constructor() MyHashMap {
	var myHashMap MyHashMap
	myHashMap.hashmap = make([]int, MAX_LENGTH)
	for i := 0; i < MAX_LENGTH; i++ {
		myHashMap.hashmap[i] = -1
	}
	return myHashMap
}

func (this *MyHashMap) Put(key int, value int) {
	this.hashmap[key] = value
}

func (this *MyHashMap) Get(key int) int {
	return this.hashmap[key]
}

func (this *MyHashMap) Remove(key int) {
	this.hashmap[key] = -1
}


/**
 * Your MyHashMap object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Put(key,value);
 * param_2 := obj.Get(key);
 * obj.Remove(key);
 */